Gems/MotionMatching/Code/Source/FeatureMatrixMinMaxScaler.cpp

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Allocators.h>
+#include <fstream>
+#include <iostream>
+#include <limits>
+#include <FeatureMatrixMinMaxScaler.h>
+
+namespace EMotionFX::MotionMatching
+{
+    AZ_CLASS_ALLOCATOR_IMPL(MinMaxScaler, MotionMatchAllocator, 0)
+
+    bool MinMaxScaler::Fit(const FeatureMatrix& in, const Settings& settings)
+    {
+        const FeatureMatrix::Index numRows = in.rows();
+        const FeatureMatrix::Index numColumns = in.cols();
+
+        m_clip = settings.m_clip;
+        m_featureMin = settings.m_featureMin;
+        m_featureMax = settings.m_featureMax;
+        m_featureRange = settings.m_featureMax - settings.m_featureMin;
+        AZ_Assert(m_featureRange > s_epsilon, "Feature range too small. This will lead to divisions by infinity.");
```

Comment:
Wouldn't it be divisions by zero (or very small values)?

Reply:
My bad, ofc yes.